File: OtherAlgorithms/Fibonacci.cpp
#include <iostream>

using namespace std;

class Fibonacci {
	public:
		int fib(int seq) {

			if (seq <= 1) {
				return seq; 
			} else {
				return (fib(seq - 1) + fib(seq - 2));
			}

			
		}		
};

int main() {
	Fibonacci fibo;

	cout << fibo.fib(10) << endl;
}

File: OtherAlgorithmsImpl/Fibonacci.java

public class Fibonacci{

	Fibonacci() {}

	public static int fib(int num) {
		
		if (num <= 1) {
			return num;
		} else {
			return (fib(num - 1) + fib(num - 2));	
		} 
	}

	public static void main(String[] args) {
		System.out.println(fib(4));		
	}
}
